src/java.base/share/classes/java/security/MessageDigest.java

```diff
@@ -457,10 +457,10 @@ public String toString() {
      * the same length and all bytes at corresponding positions are equal.
      *
      * @implNote
-     * All bytes in {@code digesta} are examined to determine equality.
-     * The calculation time depends only on the length of {@code digesta}.
-     * It does not depend on the length of {@code digestb} or the contents
-     * of {@code digesta} and {@code digestb}.
+     * All bytes in {@code digesta} are examined to determine equality, unless
+     * {@code digestb} is null or has a length of zero bytes. If {@code digestb}
+     * is not {@code null} and does not have a length of zero bytes, then the
+     * calculation time depends only on the length of {@code digesta}.
```

Comment:
Sorry, my last comment may have been not specific enough but I don't think you should remove the last sentence - I think that is still important to explain how it is implemented:

"It does not depend on the length of {@code digestb} or the contents of {@code digesta} and {@code digestb}."